rzup/src/components.rs

```diff
@@ -0,0 +1,284 @@
+    }
+    Ok(())
+}
+
+#[cfg(not(feature = "install"))]
+fn extract_archive(_env: &Environment, _archive_path: &Path, _target_dir: &Path) -> Result<()> {
+    Err(RzupError::Other("not built with install support".into()))
+}
+
+pub fn install(
+    component: &Component,
+    env: &Environment,
+    base_urls: &BaseUrls,
+    version: &Version,
+    force: bool,
+) -> Result<()> {
+    let component_to_install = component.parent_component().unwrap_or(*component);
+
+    let distribution = GithubRelease::new(base_urls);
+
+    env.emit(RzupEvent::InstallationStarted {
+        id: component.to_string(),
+        version: version.to_string(),
+    });
+
+    let archive_name = distribution.get_archive_name(&component_to_install, env.platform())?;
+    let downloaded_file = env.tmp_dir().join(archive_name);
+
+    if force {
+        Paths::cleanup_version(env, &component_to_install, version)?;
+    }
+
+    // Download and extract
+    distribution.download_version(env, &component_to_install, version)?;
+    let version_dir = Paths::get_version_dir(env, &component_to_install, version);
+
+    if let Err(e) = extract_archive(env, &downloaded_file, &version_dir) {
+        Paths::cleanup_version(env, &component_to_install, version)?;
+        return Err(e);
+    }
+
+    if let Err(e) = std::fs::remove_file(&downloaded_file) {
+        env.emit(RzupEvent::Debug {
+            message: format!("Failed to remove downloaded archive: {e}"),
+        });
+    }
+
+    env.emit(RzupEvent::InstallationCompleted {
+        id: component.to_string(),
+        version: version.to_string(),
+    });
+
+    Ok(())
+}
+
+#[cfg(feature = "install")]
+fn symlink(original: &Path, link: &Path) -> Result<()> {
+    if std::fs::symlink_metadata(link).is_ok() {
+        std::fs::remove_file(link)?;
+    }
+    if let Some(parent) = link.parent() {
+        std::fs::create_dir_all(parent)?;
+    }
+    std::os::unix::fs::symlink(original, link)
+        .map_err(|e| RzupError::Other(format!("Failed to create symlink: {e}")))
+}
+
+#[cfg(not(feature = "install"))]
+fn symlink(_original: &Path, _link: &Path) -> Result<()> {
+    Err(RzupError::Other("not built with install support".into()))
+}
+
+pub fn set_default(env: &Environment, component: &Component, version: &Version) -> Result<()> {
+    let installed_component = component.parent_component().unwrap_or(*component);
+    let version_dir = Paths::get_version_dir(env, &installed_component, version);
+
+    match component {
+        Component::CargoRiscZero => symlink(
+            &version_dir.join("cargo-risczero"),
+            &env.cargo_bin_dir().join("cargo-risczero"),
+        )?,
+        Component::R0Vm => symlink(&version_dir.join("r0vm"), &env.cargo_bin_dir().join("r0vm"))?,
+        Component::RustToolchain => {
+            symlink(&version_dir, &env.rustup_toolchain_dir().join("risc0"))?
+        }
+        Component::CppToolchain => symlink(
+            &version_dir.join("riscv32im-linux-x86_64"),
```

Comment:
The hardcoded path `riscv32im-linux-x86_64` doesn't match the platform-specific paths we create, causing this symlink to fail on non-x86 systems

Reply:
oh yeah, thats a good catch. I need to fill out test coverage for lots of things with a different platform I think